syntax = "proto3";

service GPTService {
    rpc GetAnswerWithGPT (GPTRequest) returns (GPTReply) {}
}

message Exhibit {
    string label = 1;
    repeated string descriptions = 2;
}

message GPTRequest {
    string user_question = 1;
    repeated Exhibit exhibits = 2;
    string museum_name = 3;
}

message GPTReply {
    string prompt = 1;
    string completion = 2;
    int32 prompt_tokens = 3;
    int32 completion_tokens = 4;
}